drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,455 @@
+/* sensor_apds9960.c - driver for APDS9960 ALS/RGB/gesture/proximity sensor */
+
+/*
+ *
+ *Copyright (c) 2017 Intel Corporation
+ *
+ *SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include "apds9960.h"
+
+static int apds9960_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct apds9960_data *data = dev->driver_data;
+	u8_t cmsb, clsb, rmsb, rlsb, blsb, bmsb, glsb, gmsb;
+	u8_t pdata;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	/* Read lower byte following by MSB  Ref: Datasheet : RGBC register */
+	if (i2c_reg_read_byte(data->i2c, APDS9960_I2C_ADDRESS,
```

Comment:
All of these calls to i2c_ functions return errno's which should be returned rather than be replaced with EIO